flaskr/manage.py

```diff
@@ -16,5 +17,94 @@ def make_shell_context():
+        if method_call_seq is None:
+            method_call_seq = method.find('div', class_='method-heading').text
+            args = method.find('div', class_='method-args')
+            if args is not None:
+                method_call_seq += args
+            method_call_seq
+        else:
+            method_call_seq = concat_strings(method_call_seq)
+        method_descriptions = method.find_all('p')
+        method_description = concat_strings(method_descriptions)
+        code_snippet = method.find('pre', class_='ruby')
+        if code_snippet is None:
+            code_snippet = ''
+        else:
+            code_snippet = code_snippet.text
+        method = Method(name=method_name, docs_url=url, syntax=method_call_seq, description=method_description, snippet=code_snippet, language=language)
+        db.session.add(method)
+        db.session.commit()
+
+@manager.command
+def get_js_methods():
+    language = Language.query.filter_by(name='JavaScript').first()
+    if language is None:
+        language = Language(name='JavaScript')
+        db.session.add(language)
+        db.session.commit()
+    URL = 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array'
+    page = requests.get(URL)
+    soup = BeautifulSoup(page.content, 'html.parser')
+    methods_div = soup.find('div', class_="quick-links")
+    methods_list = methods_div.find('ol')
+    method_links = methods_list.find_all('a')
+    for link in method_links[6:45]:
+        description = link['title']
+        if "deprecated" not in description:
+            name = link.find('code').text
+            url = 'https://developer.mozilla.org/' + link['href']
+            method_page = requests.get(url)
+            method_soup = BeautifulSoup(method_page.content, 'html.parser')
+            call_seq_box = method_soup.find('pre', class_='syntaxbox') #strip this?
+            if call_seq_box is not None:
+                call_seq = call_seq_box.text
+            else:
+                call_seq = method_soup.find('pre').text
+
+            edit_box = method_soup.find('iframe')
+            if edit_box is not None:
+                code_page = requests.get(edit_box['src'])
+                code_soup = BeautifulSoup(code_page.content, 'html.parser')
+
+                snippet_box = code_soup.find('pre')
+                if snippet_box is not None:
+                    snippet = snippet_box.text
+                else:
+                    example_header = method_soup.find('h2', string='Examples')
+                    first_example = example_header.find_next_sibling('h3')
+                    snippet = first_example.find_next_sibling('pre').text
+            else:
+                snippet = method_soup.find('pre').text
+            method = Method(name=name, docs_url=url, syntax=call_seq, description=description, snippet=snippet, language=language)
+            db.session.add(method)
+            db.session.commit()
+
```

Comment:
Would it be possible to move all of this into a separate file? Maybe create a popo and put everything in a class method?

Reply:
Probably at some point. I discussed this with Josh and also worked on this with Daniel before making the PR if it makes sense to try to refactor it now or put it in as is because it's working and work on a refactor later. The consensus was to move forward as is for now.

Reply:
I would like to learn how to do this in general but not clear on how to inject the dependencies we need (specifically the DB instance). I would be up for coming back and refactoring this (and the model definitions) later after we have a chance to learn about that.